extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -396,6 +417,216 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
+			buttonArialLabel: localizedConstants.RemoveButton,
+			buttonHandler: () => this.onRemoveDatabaseFilesButtonClicked()
+		};
+		const editbuttonComponent: DialogButtonComponent = {
+			buttonArialLabel: localizedConstants.EditButton,
+			buttonHandler: (button) => this.onEditDatabaseFilesButtonClicked(button)
+		};
+		const databaseFilesButtonContainer = this.addButtonsForTable(this.databaseFilesTable, addButtonComponent, removeButtonComponent, editbuttonComponent);
+
+		return this.createGroup(localizedConstants.DatabaseFilesText, [this.databaseFilesTable, databaseFilesButtonContainer], true);
+	}
+
+	/**
+	 * Converts the file object to a data view object
+	 * @param file database file object
+	 * @returns data view object
+	 */
+	private convertToDataView(file: DatabaseFile): any[] {
+		return [
+			file.name,
+			file.type,
+			file.fileGroup,
+			file.sizeInMb,
+			file.isAutoGrowthEnabled ? localizedConstants.AutoGrowthValueStringGenerator(file.type !== localizedConstants.FilestreamFileType
+				, file.autoFileGrowth.toString()
+				, file.autoFileGrowthType === FileGrowthType.Percent
+				, file.maxSizeLimitInMb) : localizedConstants.NoneText,
+			file.path,
+			file.fileNameWithExtension
+		];
+	}
+
+	private async onAddDatabaseFilesButtonClicked(button: azdata.ButtonComponent): Promise<void> {
+		// Open file dialog to create file
+		const result = await this.openDatabaseFileDialog(button);
+		if (!isUndefinedOrNull(result)) {
+			this.objectInfo.files?.push(result);
+			var newData = this.objectInfo.files?.map(file => {
+				return this.convertToDataView(file);
+			});
+			await this.setTableData(this.databaseFilesTable, newData, DefaultMaxTableRowCount)
+		}
+	}
+
+	private async onEditDatabaseFilesButtonClicked(button: azdata.ButtonComponent): Promise<void> {
+		if (this.databaseFilesTable.selectedRows.length === 1) {
+			const result = await this.openDatabaseFileDialog(button);
+			if (!isUndefinedOrNull(result)) {
+				this.objectInfo.files[this.databaseFilesTable.selectedRows[0]] = result;
+				var newData = this.objectInfo.files?.map(file => {
+					return this.convertToDataView(file);
+				});
+				await this.setTableData(this.databaseFilesTable, newData, DefaultMaxTableRowCount)
+			}
+		}
+	}
+
+	/**
+	 * Removes the selected database file from the table
+	 */
+	private async onRemoveDatabaseFilesButtonClicked(): Promise<void> {
+		if (this.databaseFilesTable.selectedRows.length === 1) {
+			this.objectInfo.files?.splice(this.databaseFilesTable.selectedRows[0], 1);
+			var newData = this.objectInfo.files?.map(file => {
+				return this.convertToDataView(file);
+			});
+			await this.setTableData(this.databaseFilesTable, newData, DefaultMaxTableRowCount)
+		}
+	}
+
+	/**
+	 * Validate the selected row to enable/disable the remove button
+	 * @returns true if the remove button should be enabled, false otherwise
+	 */
+	protected override removeButtonOnRowSelected(): boolean {
```

Comment:
```suggestion
	protected override get removeButtonEnabled(): boolean {
```
Using "OnRowSelected" here makes it seem like the method returns an Event. You can also turn this into a getter to simplify calling it, since no arguments are needed.